components/AvalancheForecast.tsx

```diff
@@ -42,7 +42,7 @@ export const AvalancheForecast: React.FunctionComponent<AvalancheForecastProps>
         navigation.setOptions({title: thisZone.name});
       }
     }
-  }, [forecast]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [forecast, forecast_zone_id, navigation]);
```

Comment:
Please let's not do this unless we have a good reason - these are super tricky and the lint rule for this particular thing in my experience is really not correct.

Reply:
I guess to expand on these - I spent time carefully choosing what went into the dependency array since it's so critical for correctness & performance and at least in the past I've found that following this linter is a good way to get infinite re-renders and ill defined behavior. I'm of course not an expert but if we change these I would just hope that the change is motivated by carefully considering the hook vs a linter rule applied broadly. Perhaps in some of these cases `useCallback` or `useMemo` might be needed as well for effects capturing functions, and as in this case changing `forecast_zone_id` would cause a full re-render for the component anyway.

Reply:
Hmm, I'm pretty surprised to hear that - we had this rule enabled on a fairly large React codebase at Stripe and I don't recall seeing it suppressed at all! I'm sure there can be weird cases, but if the callback is firing more than it should be, then it might mean the component is rendering more than it should be as well, since these inputs are ultimately derived from props & state.

I can leave these in for now and create issues to look more deeply at each of them later, but I personally would rather have the warning than not ([Dan Abramov agrees FWIW](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528) 😄 ).

Reply:
Sounds good! Happy to remove them if we're pretty certain it's a good thing. I vaguely remember struggling with cyclical re-render because I followed the linter rules in the summer, but it's more than likely I was just messing something else up entirely. If you have not had poor experiences with this in the past, it's more likely just me being a React noob - when you made the changes in this PR did anything jump out at you as obviously weird for this lint rule?

Reply:
I'd say this one felt unsafe to me, thinking through how the callback would behave if `forecast_zone_id` or the `navigation` object changed on render pass 2, and then `forecast` changed on render pass 3. At that point the callback would execute with stale references in the closure and probably not do what you want.

I think in practice that would never happen and it's probably a wash, but generally I think it's great to have the warning. It makes you consider everything that the callback is closing over.

Reply:
Opened https://github.com/stevekuznetsov/avalanche-forecast/issues/16 to track following up on this

Reply:
Sounds good!